treeherder/push_health/tests.py

```diff
@@ -164,7 +162,38 @@ def has_line(failure_line, log_line_list):
     )
 
 
-def get_test_failures(push, parent_push=None):
+def get_test_failure_jobs(push):
+    testfailed_jobs = (
+        Job.objects.filter(push=push, tier__lte=2, result='testfailed',)
+        .exclude(Q(machine_platform__platform='lint') | Q(job_type__symbol='mozlint'),)
+        .select_related('machine_platform', 'taskcluster_metadata')
```

Comment:
I am unable to see why should be add job_type to the select-related. I already see ```job_type_name``` in the jobs parameter
Each job object looks mostly like this
```"test-android-hw-g5-7-0-arm7-api-16-shippable/opt-browsertime-tp6m-geckoview-cold-wikipedia-e10s": [
            {
                "id": 21302,
                "machine_platform_id": 67,
                "option_collection_hash": "102210fe594ee9b33d82058545b1ed14f4c8206e",
                "job_type_id": 3586,
                "job_group_id": 26,
                "result": "testfailed",
                "state": "completed",
                "failure_classification_id": 1,
                "push_id": 1683,
                "start_time": "2020-07-27T17:22:36",
                "job_type_name": "test-android-hw-g5-7-0-arm7-api-16-shippable/opt-browsertime-tp6m-geckoview-cold-wikipedia-e10s",
                "job_type_symbol": "wikipedia-c",
                "platform": "android-hw-g5-7-0-arm7-api-16-shippable",
                "task_id": "evxdAiHITiG8n5f2AD5Gzw",
                "run_id": 0
            }
        ],
```

Also, I tested it previously for only health . forgot for health_summary
It works fine now

Reply:
`select_related` is a Django optimization for making queries run faster.  In some cases, if you don't use it when you are accessing a `ForeignKey` value, then if you reference it (like `job_type.name` then it will have to query the `JobType` model/table in order to get that name.  And it has to do a tiny query for EACH record you do it on.  It can be quite slow...  So with `select_related` it does a single query for ALL the records to get the `ForeignKey` data.  It will still work without it, but be slower.

Reply:
Thanks a lot for explaining me.
I will just make the changes and ping you so that you review it today

Reply:
Done this change @camd . seems fine to have another review

Reply:
I was just trying the branch with this push:

http://localhost:5000/pushhealth.html?repo=try&revision=56768a05f202768ca12851748056de435dd89802

And I got an error in the Docker container:

            | Traceback (most recent call last):
            |   File "/usr/local/lib/python3.7/site-packages/django/core/handlers/exception.py", line 34, in inner
            |     response = get_response(request)
            |   File "/usr/local/lib/python3.7/site-packages/django/core/handlers/base.py", line 115, in _get_response
            |     response = self.process_exception_by_middleware(e, request)
            |   File "/usr/local/lib/python3.7/site-packages/django/core/handlers/base.py", line 113, in _get_response
            |     response = wrapped_callback(request, *callback_args, **callback_kwargs)
            |   File "/usr/local/lib/python3.7/site-packages/django/views/decorators/csrf.py", line 54, in wrapped_view
            |     return view_func(*args, **kwargs)
            |   File "/usr/local/lib/python3.7/site-packages/rest_framework/viewsets.py", line 114, in view
            |     return self.dispatch(request, *args, **kwargs)
            |   File "/usr/local/lib/python3.7/site-packages/rest_framework/views.py", line 505, in dispatch
            |     response = self.handle_exception(exc)
            |   File "/usr/local/lib/python3.7/site-packages/rest_framework/views.py", line 465, in handle_exception
            |     self.raise_uncaught_exception(exc)
            |   File "/usr/local/lib/python3.7/site-packages/rest_framework/views.py", line 476, in raise_uncaught_exception
            |     raise exc
            |   File "/usr/local/lib/python3.7/site-packages/rest_framework/views.py", line 502, in dispatch
            |     response = handler(request, *args, **kwargs)
            |   File "/app/treeherder/webapp/api/push.py", line 262, in health
            |     push_health_test_failures = get_test_failures(push, jobs, parent_push,)
            |   File "/app/treeherder/push_health/tests.py", line 230, in get_test_failures
            |     parent_test_failures = get_test_failures(parent_push, jobs)
            |   File "/app/treeherder/push_health/tests.py", line 217, in get_test_failures
            |     push_failures = get_current_test_failures(push, option_map, jobs)
            |   File "/app/treeherder/push_health/tests.py", line 116, in get_current_test_failures
            |     countPassed = len(list(filter(lambda x: x['result'] == 'success', jobs[job_name])))
            | KeyError: 'test-macosx1014-64/debug-web-platform-tests-backlog-e10s-2'

Would you take another look?  Not quite sure why it's hitting the key error.  But you may need some kind of fallback for `jobs[job_name]` in case `jobs` doens't have that key.

Reply:
I did a terrible mistake out there :sweat:  
I passed the jobs meant for the original push to the parent push
Are the jobs for the parent push also needed in the API or should I just fetch them, use them in the backend and not show them on API

Reply:
I did't have all the revisions of the parent push on my local and still works on my local machine. Thats why , I could not catch these small bugs
Currently, I have made the changes, but I it may still break on the actual server because I can't test things to my best currently

Reply:
> I did a terrible mistake out there 😓
> I passed the jobs meant for the original push to the parent push
> Are the jobs for the parent push also needed in the API or should I just fetch them, use them in the backend and not show them on API

No, we don't need the jobs for the parent push at all.  :)

Reply:
@SuyashSalampuria : I'm hitting an odd issue with retriggering.  Do you have access to retrigger on try?   This may have something to do with running it locally.  So I will push your branch up to prototype to test to be sure it works ok there.  I get an error:

    code | "unauthorized_client"
    name | "AuthorizationError"

Reply:
OK, so retriggers work well on prototype.  So it's something with the authentication on the Docker container, I think.  I'll file a bug.  But the good news is that this branch looks about ready to be merged.  :)  I'll do one final review and let you know.